### PR TITLE
Fix validators list in extra

### PR DESCRIPTION
**Pull request by b00ris**
There were not only validators users in genesis extra field.
We need only validators.

### CODE REVIEW

File: contracts/autonity/autonity.go [hunk 1]
```diff
@@ -89,11 +90,11 @@ func (ac *Contract) DeployAutonityContract(chain consensus.ChainReader, header *
 	sender := vm.AccountRef(chain.Config().AutonityContractConfig.Deployer)
 
 	//todo do we need it?
-	//validators, err = ac.SavedValidatorsRetriever(1)
-	//sort.Sort(validators)
+	//users, err = ac.SavedValidatorsRetriever(1)
```
Comment (JekaMas) [inline contracts/autonity/autonity.go]: Could we remove commented code?

File: contracts/autonity/autonity.go [hunk 2]
```diff
@@ -102,20 +103,20 @@ func (ac *Contract) DeployAutonityContract(chain consensus.ChainReader, header *
 		return common.Address{}, err
 	}
 
-	ln := len(chain.Config().AutonityContractConfig.GetValidatorUsers())
-	validators := make(common.Addresses, 0, ln)
+	ln := len(chain.Config().AutonityContractConfig.Users)
```
Comment (JekaMas) [inline contracts/autonity/autonity.go]: Could you put here a comment why and where we should use Users and Validators?

Comment (b00ris) [reply contracts/autonity/autonity.go]: added. But this case is just removed useless allocation.

**Comment by yazzaoui** [comment]
can you check please if https://github.com/clearmatics/autonity/pull/318/commits/3422d79a636d054ba04446881eb7e5451023e141 is not fixing this issue?

**Comment by b00ris** [comment]
Yes, it's the same. But my pr has one more change and comments.

File: contracts/autonity/autonity.go [hunk 3]
```diff
@@ -134,34 +135,27 @@ func (ac *Contract) DeployAutonityContract(chain consensus.ChainReader, header *
...
+	users := make(common.Addresses, 0, ln)
 	enodes := make([]string, 0, ln)
 	accTypes := make([]*big.Int, 0, ln)
 	participantStake := make([]*big.Int, 0, ln)
-	for _, v := range chain.Config().AutonityContractConfig.Users { // Todo: Looping through all the users here
-		validators = append(validators, v.Address) //Todo: Is this appending all the users as validators?
+	for _, v := range chain.Config().AutonityContractConfig.Users {
```
Comment (aesmonty) [inline contracts/autonity/autonity.go]: I think there is still a problem here. In the Extradata, there should be only validators, and here we are adding all users, right? @b00ris

**Pull request merged**